ksqldb-engine/src/main/java/io/confluent/ksql/query/KafkaStreamsQueryValidator.java

```diff
@@ -69,10 +77,41 @@ private void validateCacheBytesUsage(
       return;
     }
     final long configured = getCacheMaxBytesBuffering(config);
-    final long usedByRunning = running.stream()
-        .mapToLong(r -> new StreamsConfig(r.getStreamsProperties())
-                .getLong(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG))
-        .sum();
+    long usedByRunning;
+    if (!config.getConfig(true).getBoolean(KsqlConfig.KSQL_SHARED_RUNTIME_ENABLED)) {
+      usedByRunning = running.stream()
+          .mapToLong(r -> new StreamsConfig(r.getStreamsProperties())
+              .getLong(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG))
+          .sum();
+    } else {
+      usedByRunning = running.stream()
+          .filter(t -> !(t instanceof BinPackedPersistentQueryMetadataImpl))
+          .mapToLong(r -> new StreamsConfig(r.getStreamsProperties())
+              .getLong(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG))
+          .sum();
+      final Set<String> runtimes = new HashSet<>();
+      Long maxCache = -1L;
+      for (final QueryMetadata queryMetadata: running) {
+        if (queryMetadata instanceof BinPackedPersistentQueryMetadataImpl) {
+          if (maxCache == -1L) {
+            maxCache = (Long) queryMetadata.getStreamsProperties()
+                .get(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG);
+          }
+          if (!runtimes.contains(queryMetadata.getQueryApplicationId())) {
+            runtimes.add(queryMetadata.getQueryApplicationId());
+            usedByRunning += (Long) queryMetadata.getStreamsProperties()
+                .get(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG);
+          } else {
+            if (!Objects.equals(maxCache, (Long) queryMetadata.getStreamsProperties()
+                .get(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG))) {
+              LOG.warn("Inconsistent "
```

Comment:
nit: can we log what the two inconsistent values are? And maybe the application id of the runtime that isn't matching?